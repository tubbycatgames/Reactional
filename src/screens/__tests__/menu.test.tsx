import React from 'react'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import Menu from '../menu'

describe('Menu', () => {
  it('should render', () => {
    const navigation = { navigate: () => {} }
    const render = renderer.create(<Menu navigation={navigation} />)
    expect(render.toJSON()).toMatchSnapshot()
  })

  it('should bind a nav action on button press', () => {
    const navigate = jest.fn()
    const navigation = { navigate }
    const render = renderer.create(<Menu navigation={navigation} />)

    const button = render.root.findAllByType(Button)
    button[0].props.onPress()
    expect(navigate).toHaveBeenCalledWith('Game')
    button[1].props.onPress()
    expect(navigate).toHaveBeenCalledWith('Swipe')
  })
})
