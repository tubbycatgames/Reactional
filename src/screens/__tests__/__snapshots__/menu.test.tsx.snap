// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#B3EFFF",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Reactional!
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    An accessible, open-source game about reactions
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Meant for learning, guiding, and enjoyment
  </Text>
  <View
    accessibilityLabel="Click to play the Game"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#046B99",
            },
          ]
        }
      >
        Play!
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="Click to enter Swipe Mode"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#046B99",
            },
          ]
        }
      >
        Swipe!
      </Text>
    </View>
  </View>
</View>
`;
